src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +85,13 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<People> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getSex() == People.Sex.WOMEN
+                && p.getAge() >= femaleAge && p.getCats() != null)
+                .flatMap(p -> p.getCats()
+                .stream())
```

Comment:
these can be inlined
```suggestion
                .flatMap(p -> p.getCats().stream())
```